src/main/kotlin/com/learnspigot/bot/manager/ForumManager.kt

```diff
@@ -33,10 +34,23 @@ class ForumManager(private val bot: JDA, private val datastore: Datastore, priva
     init {
         logger.info("Initiating ForumManager")
         bot.listener<MessageReceivedEvent> {
-            if(it.author.isBot) return@listener
             val channel = it.channel
-            if(channel !is ThreadChannel) return@listener
-            if(channel.parentChannel.id != System.getenv("HELP_CHANNEL_ID")) return@listener
+            if (channel !is ThreadChannel) return@listener
+            if (channel.parentChannel.id != System.getenv("HELP_CHANNEL_ID")) return@listener
+            val threadChannel: ThreadChannel = channel
+
+            if (it.message.type == MessageType.CHANNEL_PINNED_ADD) {
+                if (it.author == bot.selfUser) {
+                    it.message.delete().queue()
+                }
+            }
+
+            if (it.author.isBot) return@listener
+
+            threadChannel.history.retrievePast(2).queue {
```

Comment:
give the variable a name as it overrides the variable of the event lambda. I would just call it `messages`.

Reply:
Ahh ok good thinking.